core/src/Kokkos_Core.hpp

```diff
@@ -144,6 +144,10 @@ void pre_initialize(const InitArguments& args);
 
 void post_initialize(const InitArguments& args);
 
+void declare_configuration_metadata(const std::string& category,
+                                    const std::string& key,
+                                    const std::string& value);
```

Comment:
Why do you need the declaration in `Kokkos_Core.hpp`?  Where else than `Kokkos_Core.cpp` would we use?

Reply:
This could be used in backends, the CUDA backend will likely declare configuration metadata as well